gkeys/gkeys/keyhandler.py

```diff
@@ -108,3 +111,27 @@ def key_search(self, args, first_match=False):
 
         self.logger.debug(_unicode("KeyHandler: key_search; keys = %s") % str(keys))
         return keys
+
+    def _is_expiring(self, results, days_limit):
+        is_expiring = False
+        for key in results:
+            for specs in results[key]:
+                if specs.days > days_limit*(-1) and specs.days < days_limit:
+                    is_expiring = True
+                    break
+        return is_expiring
+
+    def _generate_template(self, template_path, keyprints, specprint):
+        with open(template_path, 'r') as file_contents:
+            content = file_contents.read()
+        message_template = Template(content)
+        subkey_msg = ''
+        message = message_template.substitute(key_print=keyprints, spec_print=specprint)
+        return message
+
+    def _find_email(self, uid):
+        match = re.findall(r'[\w\.-]+@[\w\.-]+', uid)
+        uid = ''
+        if match:
+            uid = match[0]
+        return uid
```

Comment:
this method has no way of searching for a preferred email if that option is given.

Also is a public method, no leading _